File: frontend/src/api/rol.service.ts
// Capa de API de Roles.
import http from '@/api/http'
import type { RolDetalleDto, RolDto } from '@/types/rol.type'

export function listarRoles() {
  return http.get<RolDetalleDto[]>('/roles/listar').then((r) => r.data)
}

export function crearRol(dto: RolDto) {
  return http.post<RolDetalleDto>('/roles/crear', dto).then((r) => r.data)
}

export function actualizarRol(idRol: number, dto: RolDto) {
  return http.put<RolDetalleDto>('/roles/actualizar', dto, { params: { idRol } }).then((r) => r.data)
}

export function eliminarRol(idRol: number) {
  return http.delete('/roles/eliminar', { params: { idRol } })
}

File: src/main/java/com/uap/control_tickets/services/interfaces/RolService.java
package com.uap.control_tickets.services.interfaces;

import com.uap.control_tickets.dto.rol.RolDetalleDto;
import com.uap.control_tickets.dto.rol.RolDto;

import java.util.List;

/** Contrato del CRUD de Rol. */
public interface RolService {
    List<RolDetalleDto> listar();
    RolDetalleDto obtener(Long idRol);
    RolDetalleDto crear(RolDto dto);
    RolDetalleDto actualizar(Long idRol, RolDto dto);
    void eliminar(Long idRol);
}
